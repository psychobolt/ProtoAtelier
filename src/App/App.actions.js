export const Actions = {
  INIT_APP: 'initApp',
};

export const init = () => ({ type: Actions.INIT_APP });
