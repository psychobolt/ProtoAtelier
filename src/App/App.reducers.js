import editorReducer from './Editor/Editor.reducer';

export default {
  editor: editorReducer,
};
