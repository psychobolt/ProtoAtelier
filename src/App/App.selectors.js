export const getApp = state => state.app || {};

export const getEditor = state => getApp(state).editor || {};

export const getProject = state => getApp(state).project || {};

export const getProjectHistory = state => getProject(state).history || {};

export const getProjectWithoutHistory = state => {
  const { history, ...rest } = getProject(state);
  return rest;
};

export const getPastProject = state => getProjectHistory(state).past || {};

export const getPresentProject = state => getProjectHistory(state).present || {};

export const getFutureProject = state => getProjectHistory(state).future || {};

export const getProjectCanvas = state => getPresentProject(state).canvas || {};

export const getWorldProject = state => getPresentProject(state).world || {};

export const getCanvasProject = state => getProject(state).config || {};

export const getSettings = state => getProject(state).settings || {};

export const getCanvasSettings = state => getSettings(state).canvas || {};

export const getWorldSettings = state => getSettings(state).world || {};

export const getWorkspace = state => ({
  ...getProjectWithoutHistory(state),
  project: getPresentProject(state),
});

export const getCanvas = state => ({
  ...getEditor(state).canvas,
  ...getProjectCanvas(state),
  ...getCanvasSettings(state),
});

export const getWorld = state => ({
  ...getEditor(state).world,
  ...getWorldProject(state),
  ...getWorldSettings(state),
});
