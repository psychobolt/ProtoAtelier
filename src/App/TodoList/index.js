export { default as TodoFilter, Filters } from './TodoFilter';
export { default as TodoForm } from './TodoForm';
export { default } from './TodoList.container';
