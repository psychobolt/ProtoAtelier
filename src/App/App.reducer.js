import editorReducer from './Editor/Editor.reducer';
import projectReducer from './Project/Project.reducer';
import { Actions as ProjectActions } from './Project/Project.actions';
import { Actions as AppActions } from './App.actions';
import { getProject } from './App.selectors';

export default (state = {}, action) => {
  switch (action.type) {
    case AppActions.INIT_APP: return {};
    default: {
      const { type } = action;
      if (ProjectActions.NEW_PROJECT === type
          || ProjectActions.LOAD_PROJECT === type
          || getProject({ app: state }).id) {
        return {
          ...state,
          editor: editorReducer(state.editor, action),
          project: projectReducer(state.project, action),
        };
      }
      return state;
    }
  }
};
