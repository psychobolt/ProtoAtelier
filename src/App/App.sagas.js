import { all, put, fork } from 'redux-saga/effects';
import { router } from 'redux-saga-router';

import editorSaga from './Editor/Editor.sagas';
import projectSaga from './Project/Project.sagas';
import { init } from './App.actions';

export function* onRoot() {
  yield put(init());
}

export default function* saga(history) {
  yield all([
    fork(router, history, {
      '/': onRoot,
      // '/project/:id?': onProject, TODO
    }),
    projectSaga(),
    editorSaga(),
  ]);
}
