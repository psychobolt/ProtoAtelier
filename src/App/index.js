import { hot } from 'react-hot-loader';

import App from './App.component';

export { default as initialState } from './App.state';

export default hot(module)(App);
