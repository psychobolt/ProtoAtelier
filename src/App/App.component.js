// @flow
import React from 'react';
import type { Match } from 'react-router-dom';
import styled from 'styled-components';

import TodoList, { TodoForm, TodoFilter, Filters } from './TodoList';
import * as styles from './App.style';

const Container = styled.div`${styles.container}`;
const List = TodoList.extend`${styles.todoList}`;
const Form = TodoForm.extend`${styles.todoForm}`;
const Filter = TodoFilter.extend`${styles.todoFilter}`;

const App = ({ match }: { match: Match }) => (
  <Container>
    <Form />
    <List filter={match.params.filter || Filters.ALL}>
      <Filter centered />
    </List>
  </Container>
);

export default App;
