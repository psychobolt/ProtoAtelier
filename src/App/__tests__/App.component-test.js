import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import { Filters } from '../TodoList';
import App from '../App.component';

const mockStore = configureMockStore([]);

describe('component <App />', () => {
  [Filters.ALL, undefined].forEach(filter => {
    it(` should render correctly with filter ${filter}`, () => {
      const props = {
        match: {
          params: { filter },
        },
      };
      const store = mockStore({ todos: [] });
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter keyLength={0}>
            <App {...props} />
          </MemoryRouter>
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
