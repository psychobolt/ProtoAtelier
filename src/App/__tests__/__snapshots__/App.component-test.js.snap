// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <App />  should render correctly with filter All 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <App
        match={
          Object {
            "params": Object {
              "filter": "All",
            },
          }
        }
      >
        <div
          style={
            Object {
              "margin": "auto",
              "paddingTop": "10px",
              "width": "500px",
            }
          }
        >
          <Connect(TodoForm)
            style={
              Object {
                "margin": "0 auto 10px auto",
                "width": "232px",
              }
            }
          >
            <TodoForm
              inputValue=""
              onValueSubmit={[Function]}
              style={
                Object {
                  "margin": "0 auto 10px auto",
                  "width": "232px",
                }
              }
            >
              <x-box
                style={
                  Object {
                    "margin": "0 auto 10px auto",
                    "width": "232px",
                  }
                }
              >
                <XInput
                  fallback={[Function]}
                  onKeyup={[Function]}
                  style={
                    Object {
                      "marginRight": "6px",
                    }
                  }
                  value=""
                >
                  <x-input
                    style={
                      Object {
                        "marginRight": "6px",
                      }
                    }
                  >
                    <input
                      defaultValue=""
                      onKeyUp={[Function]}
                    />
                  </x-input>
                </XInput>
                <x-button
                  onClick={[Function]}
                >
                  <x-label>
                    Add Todo
                  </x-label>
                </x-button>
              </x-box>
            </TodoForm>
          </Connect(TodoForm)>
          <Connect(TodoList)
            filter="All"
            style={
              Object {
                "margin": "10px 0",
              }
            }
          >
            <TodoList
              filter="All"
              onTodoClick={[Function]}
              style={
                Object {
                  "margin": "10px 0",
                }
              }
              todos={Array []}
            >
              <x-card
                style={
                  Object {
                    "margin": "10px 0",
                  }
                }
              >
                <TodoFilter
                  centered={true}
                  style={
                    Object {
                      "marginTop": "10px",
                    }
                  }
                >
                  <x-tabs
                    centered={true}
                    style={
                      Object {
                        "marginTop": "10px",
                      }
                    }
                  >
                    <withRouter(Connect(XTab))
                      filter="All"
                    >
                      <Route
                        render={[Function]}
                      >
                        <Connect(XTab)
                          filter="All"
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "canGo": [Function],
                              "createHref": [Function],
                              "entries": Array [
                                Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                              ],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "index": 0,
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                        >
                          <XTab
                            active={true}
                            dispatch={[Function]}
                            filter="All"
                            history={
                              Object {
                                "action": "POP",
                                "block": [Function],
                                "canGo": [Function],
                                "createHref": [Function],
                                "entries": Array [
                                  Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  },
                                ],
                                "go": [Function],
                                "goBack": [Function],
                                "goForward": [Function],
                                "index": 0,
                                "length": 1,
                                "listen": [Function],
                                "location": Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                                "push": [Function],
                                "replace": [Function],
                              }
                            }
                            location={
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              }
                            }
                            match={
                              Object {
                                "isExact": true,
                                "params": Object {},
                                "path": "/",
                                "url": "/",
                              }
                            }
                            onClick={[Function]}
                          >
                            <x-tab
                              onClick={[Function]}
                              selected={true}
                            >
                              <x-label>
                                All
                              </x-label>
                            </x-tab>
                          </XTab>
                        </Connect(XTab)>
                      </Route>
                    </withRouter(Connect(XTab))>
                    <withRouter(Connect(XTab))
                      filter="active"
                    >
                      <Route
                        render={[Function]}
                      >
                        <Connect(XTab)
                          filter="active"
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "canGo": [Function],
                              "createHref": [Function],
                              "entries": Array [
                                Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                              ],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "index": 0,
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                        >
                          <XTab
                            active={false}
                            dispatch={[Function]}
                            filter="active"
                            history={
                              Object {
                                "action": "POP",
                                "block": [Function],
                                "canGo": [Function],
                                "createHref": [Function],
                                "entries": Array [
                                  Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  },
                                ],
                                "go": [Function],
                                "goBack": [Function],
                                "goForward": [Function],
                                "index": 0,
                                "length": 1,
                                "listen": [Function],
                                "location": Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                                "push": [Function],
                                "replace": [Function],
                              }
                            }
                            location={
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              }
                            }
                            match={
                              Object {
                                "isExact": true,
                                "params": Object {},
                                "path": "/",
                                "url": "/",
                              }
                            }
                            onClick={[Function]}
                          >
                            <x-tab
                              onClick={[Function]}
                              selected={null}
                            >
                              <x-label>
                                Active
                              </x-label>
                            </x-tab>
                          </XTab>
                        </Connect(XTab)>
                      </Route>
                    </withRouter(Connect(XTab))>
                    <withRouter(Connect(XTab))
                      filter="completed"
                    >
                      <Route
                        render={[Function]}
                      >
                        <Connect(XTab)
                          filter="completed"
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "canGo": [Function],
                              "createHref": [Function],
                              "entries": Array [
                                Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                              ],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "index": 0,
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                        >
                          <XTab
                            active={false}
                            dispatch={[Function]}
                            filter="completed"
                            history={
                              Object {
                                "action": "POP",
                                "block": [Function],
                                "canGo": [Function],
                                "createHref": [Function],
                                "entries": Array [
                                  Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  },
                                ],
                                "go": [Function],
                                "goBack": [Function],
                                "goForward": [Function],
                                "index": 0,
                                "length": 1,
                                "listen": [Function],
                                "location": Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                                "push": [Function],
                                "replace": [Function],
                              }
                            }
                            location={
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              }
                            }
                            match={
                              Object {
                                "isExact": true,
                                "params": Object {},
                                "path": "/",
                                "url": "/",
                              }
                            }
                            onClick={[Function]}
                          >
                            <x-tab
                              onClick={[Function]}
                              selected={null}
                            >
                              <x-label>
                                Completed
                              </x-label>
                            </x-tab>
                          </XTab>
                        </Connect(XTab)>
                      </Route>
                    </withRouter(Connect(XTab))>
                  </x-tabs>
                </TodoFilter>
                <ul
                  style={
                    Object {
                      "listStyleType": "none",
                    }
                  }
                >
                  <x-label>
                    No items
                  </x-label>
                </ul>
              </x-card>
            </TodoList>
          </Connect(TodoList)>
        </div>
      </App>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`component <App />  should render correctly with filter undefined 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <App
        match={
          Object {
            "params": Object {
              "filter": undefined,
            },
          }
        }
      >
        <div
          style={
            Object {
              "margin": "auto",
              "paddingTop": "10px",
              "width": "500px",
            }
          }
        >
          <Connect(TodoForm)
            style={
              Object {
                "margin": "0 auto 10px auto",
                "width": "232px",
              }
            }
          >
            <TodoForm
              inputValue=""
              onValueSubmit={[Function]}
              style={
                Object {
                  "margin": "0 auto 10px auto",
                  "width": "232px",
                }
              }
            >
              <x-box
                style={
                  Object {
                    "margin": "0 auto 10px auto",
                    "width": "232px",
                  }
                }
              >
                <XInput
                  fallback={[Function]}
                  onKeyup={[Function]}
                  style={
                    Object {
                      "marginRight": "6px",
                    }
                  }
                  value=""
                >
                  <x-input
                    style={
                      Object {
                        "marginRight": "6px",
                      }
                    }
                  >
                    <input
                      defaultValue=""
                      onKeyUp={[Function]}
                    />
                  </x-input>
                </XInput>
                <x-button
                  onClick={[Function]}
                >
                  <x-label>
                    Add Todo
                  </x-label>
                </x-button>
              </x-box>
            </TodoForm>
          </Connect(TodoForm)>
          <Connect(TodoList)
            filter="All"
            style={
              Object {
                "margin": "10px 0",
              }
            }
          >
            <TodoList
              filter="All"
              onTodoClick={[Function]}
              style={
                Object {
                  "margin": "10px 0",
                }
              }
              todos={Array []}
            >
              <x-card
                style={
                  Object {
                    "margin": "10px 0",
                  }
                }
              >
                <TodoFilter
                  centered={true}
                  style={
                    Object {
                      "marginTop": "10px",
                    }
                  }
                >
                  <x-tabs
                    centered={true}
                    style={
                      Object {
                        "marginTop": "10px",
                      }
                    }
                  >
                    <withRouter(Connect(XTab))
                      filter="All"
                    >
                      <Route
                        render={[Function]}
                      >
                        <Connect(XTab)
                          filter="All"
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "canGo": [Function],
                              "createHref": [Function],
                              "entries": Array [
                                Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                              ],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "index": 0,
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                        >
                          <XTab
                            active={true}
                            dispatch={[Function]}
                            filter="All"
                            history={
                              Object {
                                "action": "POP",
                                "block": [Function],
                                "canGo": [Function],
                                "createHref": [Function],
                                "entries": Array [
                                  Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  },
                                ],
                                "go": [Function],
                                "goBack": [Function],
                                "goForward": [Function],
                                "index": 0,
                                "length": 1,
                                "listen": [Function],
                                "location": Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                                "push": [Function],
                                "replace": [Function],
                              }
                            }
                            location={
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              }
                            }
                            match={
                              Object {
                                "isExact": true,
                                "params": Object {},
                                "path": "/",
                                "url": "/",
                              }
                            }
                            onClick={[Function]}
                          >
                            <x-tab
                              onClick={[Function]}
                              selected={true}
                            >
                              <x-label>
                                All
                              </x-label>
                            </x-tab>
                          </XTab>
                        </Connect(XTab)>
                      </Route>
                    </withRouter(Connect(XTab))>
                    <withRouter(Connect(XTab))
                      filter="active"
                    >
                      <Route
                        render={[Function]}
                      >
                        <Connect(XTab)
                          filter="active"
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "canGo": [Function],
                              "createHref": [Function],
                              "entries": Array [
                                Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                              ],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "index": 0,
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                        >
                          <XTab
                            active={false}
                            dispatch={[Function]}
                            filter="active"
                            history={
                              Object {
                                "action": "POP",
                                "block": [Function],
                                "canGo": [Function],
                                "createHref": [Function],
                                "entries": Array [
                                  Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  },
                                ],
                                "go": [Function],
                                "goBack": [Function],
                                "goForward": [Function],
                                "index": 0,
                                "length": 1,
                                "listen": [Function],
                                "location": Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                                "push": [Function],
                                "replace": [Function],
                              }
                            }
                            location={
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              }
                            }
                            match={
                              Object {
                                "isExact": true,
                                "params": Object {},
                                "path": "/",
                                "url": "/",
                              }
                            }
                            onClick={[Function]}
                          >
                            <x-tab
                              onClick={[Function]}
                              selected={null}
                            >
                              <x-label>
                                Active
                              </x-label>
                            </x-tab>
                          </XTab>
                        </Connect(XTab)>
                      </Route>
                    </withRouter(Connect(XTab))>
                    <withRouter(Connect(XTab))
                      filter="completed"
                    >
                      <Route
                        render={[Function]}
                      >
                        <Connect(XTab)
                          filter="completed"
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "canGo": [Function],
                              "createHref": [Function],
                              "entries": Array [
                                Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                              ],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "index": 0,
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                        >
                          <XTab
                            active={false}
                            dispatch={[Function]}
                            filter="completed"
                            history={
                              Object {
                                "action": "POP",
                                "block": [Function],
                                "canGo": [Function],
                                "createHref": [Function],
                                "entries": Array [
                                  Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  },
                                ],
                                "go": [Function],
                                "goBack": [Function],
                                "goForward": [Function],
                                "index": 0,
                                "length": 1,
                                "listen": [Function],
                                "location": Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                                "push": [Function],
                                "replace": [Function],
                              }
                            }
                            location={
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              }
                            }
                            match={
                              Object {
                                "isExact": true,
                                "params": Object {},
                                "path": "/",
                                "url": "/",
                              }
                            }
                            onClick={[Function]}
                          >
                            <x-tab
                              onClick={[Function]}
                              selected={null}
                            >
                              <x-label>
                                Completed
                              </x-label>
                            </x-tab>
                          </XTab>
                        </Connect(XTab)>
                      </Route>
                    </withRouter(Connect(XTab))>
                  </x-tabs>
                </TodoFilter>
                <ul
                  style={
                    Object {
                      "listStyleType": "none",
                    }
                  }
                >
                  <x-label>
                    No items
                  </x-label>
                </ul>
              </x-card>
            </TodoList>
          </Connect(TodoList)>
        </div>
      </App>
    </Router>
  </MemoryRouter>
</Provider>
`;
